client/executor/src/native_executor.rs

```diff
@@ -48,8 +48,14 @@ use sp_tasks::new_async_externalities;
 use sp_version::{NativeVersion, RuntimeVersion};
 use sp_wasm_interface::{Function, HostFunctions};
 
-/// Default num of pages for the heap
-const DEFAULT_HEAP_PAGES: u64 = 2048;
+/// Default num of pages for the heap of the runtime, when used for consensus operations.
+///
+/// 128mb per instance.
+const DEFAULT_HEAP_PAGES_CONSENSUS: u64 = 2048;
+/// Default num of pages for the heap of the runtime, when used for offchain operations.
+///
+/// 256mb per instance.
+const DEFAULT_HEAP_PAGES_OFFCHAIN: u64 = DEFAULT_HEAP_PAGES_CONSENSUS * 2;
```

Comment:
Having bigger requirement for offchain is a bit awkward.
I mean most offchain process should be light (I am only aware of 'I am online', but its requirement are certainly small).
So we use a big mem profile for offchain phragmen mostly.
This makes me think that offchain worker default heap page could depend on the offchain method being call.
Same if considering overriding from cli.
(I am not sure the pool allow switching instance depending on heap page, but could also use a single use instance for this special case)